client/protocols/export.py

```diff
@@ -0,0 +1,145 @@
+from client.protocols.common import models
+from client.utils import auth
+from client.utils import network
+from datetime import datetime
+import client
+import logging
+import os
+import pickle
+import sys
+
+log = logging.getLogger(__name__)
+
+GAE_DATETIME_FORMAT = "%Y-%m-%d %H:%M:%S.%f"
+
+class ExportProtocol(models.Protocol):
+    """Downloads submissions from the server"""
+
+    def on_interact(self):
+        """If the --export parameter was used, downloads all submissions 
+        for the given assignment from the server and then exits
+        """
+        data = None
+        try:
+            data = pickle.load(open("export_cache.pkl", "rb"))
+            self.access_token = auth.authenticate(self.args.authenticate)
+            log.info('Authenticated with access token %s', self.access_token)
+        except Exception as e:
+            if self.args.export:
+                data = {}
```

Comment:
Instead of initializing `data` to an empty dictionary, just instantiate it with all of its fields after authentication:

```
self.access_token = ...
data = {
    'assign': self.args.export,
    ...
}
```